packages/credential-provider-ini/src/resolveCredentialSource.ts

```diff
@@ -13,22 +13,34 @@ import { AwsCredentialIdentityProvider } from "@smithy/types";
  */
 export const resolveCredentialSource = (
   credentialSource: string,
-  profileName: string
+  profileName: string,
+  logger?: Logger
 ): ((options?: CredentialProviderOptions) => Promise<AwsCredentialIdentityProvider>) => {
   const sourceProvidersMap = {
-    EcsContainer: (options?: CredentialProviderOptions) =>
-      import("@smithy/credential-provider-imds").then(({ fromContainerMetadata }) => fromContainerMetadata(options)),
-    Ec2InstanceMetadata: (options?: CredentialProviderOptions) =>
-      import("@smithy/credential-provider-imds").then(({ fromInstanceMetadata }) => fromInstanceMetadata(options)),
-    Environment: (options?: CredentialProviderOptions) =>
-      import("@aws-sdk/credential-provider-env").then(({ fromEnv }) => fromEnv(options)),
+    EcsContainer: async (options?: CredentialProviderOptions) => {
+      const { fromHttp } = await import("@aws-sdk/credential-provider-http");
+      const { fromContainerMetadata } = await import("@smithy/credential-provider-imds");
+      logger?.debug("@aws-sdk/credential-provider-ini - credential_source is EcsContainer");
+      return chain(fromHttp(options ?? {}), fromContainerMetadata(options));
```

Comment:
Just an observation. Its does feel a little bit jank that `EcsContainer` is being overloaded here i.e it was previously just IMDS(container meta) but now being used as IMDS OR HTTP. It would feel cleaner to define a new credential source like `FromHTTP`...

Reply:
yes, it would be out of scope to declare an entirely new ini credential source in only this SDK. In this context fromHttp is only used for its ability to connect to EKS/ECS